summarize/src/main.rs

```diff
@@ -34,6 +38,15 @@ fn main() -> Result<(), Box<std::error::Error>> {
         return Ok(());
     }
 
+    //cannot be greater than 100% or less than 0%
```

Comment:
Please give an error instead of silently capping it. It can be confusing.

Reply:
What is a good way or idiomatic way to raise error? I am writing something like this right now. 
```
    let percent_above = opt.percent_above;
    //cannot be greater than 100% or less than 0%
    if percent_above > 100.0 {
        return Err("Percentage of total time cannot be more than 100.0".into())
    } else if percent_above < 0.0 {
        return Err("Percentage of total time cannot be less than 0.0".into())
    }
```
This produces error 

```
$ ../measureme/target/release/summarize pid-22219 -p 111
Error: StringError("Percentage of total time cannot be more than 100.0")
```

Reply:
I also tried to see if `StructOpt` has a way for me to write a validator so the error handling is done at that level but maybe I missed it. I could not find a way. :fearful:

Reply:
I mean the thing with returning Err from main won't pretty print your error. We'd have to do that ourselves if we want that. You could maybe just print it with `eprintln!` and then return something that silently exists with a non-zero exit code?

Reply:
It looks like the `failure` crate [might be useful here](https://rust-lang-nursery.github.io/cli-wg/tutorial/errors.html#providing-context). I don't have any experience with it however and I don't think it's necessary to use it in this PR. 

I think for now, just use `eprintln!()` and `std::process:exit()` as @Darksonn suggests.

Reply:
Well, you should watch out with `std::process::exit`, because it doesn't run destructors of any variables in the main function. This probably doesn't matter in this case, but if you've written to files with a `BufWriter` or something, you may have buffered data that wasn't written.

I remember seeing the recommendation of having the actual `main` function just call the another "main" function with the actual code, and `std::process::exit` with the return value of that, since this way destructors in our main are run on "return" before we call `exit`.

Reply:
Highly appreciate your review and help! Given I can use process::exit() in this case, this is what I am going with.

```
    let percent_above = opt.percent_above;
    //cannot be greater than 100% or less than 0%
    if percent_above > 100.0 {
        eprintln!("Percentage of total time cannot be more than 100.0");
        std::process::exit(1);
    } else if percent_above < 0.0 {
        eprintln!("Percentage of total time cannot be less than 0.0");
        std::process::exit(1);
    }
```

Reply:
I pushed the changes. Please reopen this conversation if I have mistakenly resolved it.